data/uhc_pack/functions/pre_generation/stage/generate.mcfunction

```diff
@@ -0,0 +1,19 @@
+scoreboard players set X uhcPG 0
+scoreboard players set Z uhcPG 0
+scoreboard players set Done uhcPG 0
+scoreboard players set Total uhcPG 0
+scoreboard players set Direction uhcPG 1
+
+scoreboard players operation Total uhcPG += UHC uhcBSize
+# This is a little gross, but operation doesn't allow for using literals
+scoreboard players set ChunkSize uhcPG 16
+scoreboard players operation Total uhcPG /= ChunkSize uhcPG
+scoreboard players reset ChunkSize uhcPG
+
+scoreboard players operation Total uhcPG *= Total uhcPG
+
+bossbar add uhc_pack:pregen {"text":"Generating...","color":"white"}
```

Comment:
I would prefer that we create the bossbar once in a `create_bossbars` function similar to scoreboards, teams and storage, and simply toggle its visibility when running the generation.

Reply:
Ah yep, that'd be cleaner indeed